rust/src/storage/azure/mod.rs

```diff
@@ -126,25 +182,75 @@ fn to_storage_err(err: Box<dyn Error + Sync + std::marker::Send>) -> StorageErro
 }
 
 fn to_storage_err2(err: azure_storage::core::Error) -> StorageError {
+    if let azure_storage::core::Error::CoreError(azure_core::Error::Other(ref other_err)) = err {
```

Comment:
this does not look pretty, but unfortunately is required right now. Luckily there is ongoing work in the azure crates to "flatten" the errors...

Reply:
Yeah it definitely a pain point when making this change.  There are multiple definitions of HttpError which was confusing at first.